src/main/java/com/mraof/minestuck/alchemy/CardCaptchas.java

```diff
@@ -63,7 +71,22 @@ public static String getCaptcha(Item item, MinecraftServer server)
 	@Nullable
 	public static Item getItemFromCaptcha(String captcha, MinecraftServer server)
 	{
-		return MSExtraData.get(server).getCardCaptchas().captchasMap.inverse().get(captcha);
+		MSExtraData data = MSExtraData.get(server);
+		CardCaptchas captchas = data.getCardCaptchas();
+		
+		if(PredeterminedCardCaptchas.predefinedCardMap.inverse().containsKey(captcha) && captchas.captchasMap.inverse().containsKey(captcha))
+		{
+			captchas.captchasMap.inverse().remove(captcha);
+		}
+		
+		if(PredeterminedCardCaptchas.predefinedCardMap.inverse().containsKey(captcha))
+		{
+			return PredeterminedCardCaptchas.predefinedCardMap.inverse().get(captcha);
+		}
+		else
+		{
+			return MSExtraData.get(server).getCardCaptchas().captchasMap.inverse().get(captcha);
```

Comment:
When reaching this point, I think it'd be nice to also check for overlap of the item that was found.